feat(workflow): Adding Release Adoption Labels

iProgramStuff opened the pull request:
Adding release adoption label to FE. I didn't put this behind a feature flag because it's only present in the api if they have the feature, so I basically used it's presence to "feature flag" this.

iProgramStuff committed:
Adding adoption stage info to ReleaseSerializer

iProgramStuff committed:
Adding adoption label to FE

getsantry[bot] committed:
style(lint): Auto commit lint changes

priscilawebdev commented on static/app/views/releases/list/releaseHealth/content.tsx:
```diff
@@ -31,6 +31,7 @@ import ProjectLink from './projectLink';
 
 type Props = {
   projects: Array<ReleaseProject>;
+  adoptionStages: {} | undefined;
```
```suggestion
  adoptionStages?: Record<string, any>;
```
should we also move this prop to the bottom of this object, since it is optional?

priscilawebdev commented on static/app/views/releases/list/releaseHealth/content.tsx:
```diff
@@ -50,6 +52,11 @@ const Content = ({
   getHealthData,
 }: Props) => {
+  const adoption_map = {
+    not_adopted: 'Not Adopted',
```
should we use translations here?

this const could also be outside of this component

iProgramStuff replied:
Yeah, I wasn't sure where to put this. Was tempted to just have the backend return the text I wanted to display for this first MVP since I was coding both...but thought against it and just made the quick map here :P

iProgramStuff replied:
Okay, the backend is now sending timestamps and, in a future iteration before release, an actual front end dev will be coming in and do something better with the timestamps.

priscilawebdev commented on static/app/views/releases/list/releaseHealth/content.tsx:
```diff
@@ -150,6 +158,14 @@ const Content = ({
+                    <Column>
+                      {adoptionStages[project.slug]
+                        ? adoption_map[adoptionStages[project.slug]]
+                        : '---'}
```
Not sure if this is the case, but usually we use the component `NotAvailable` when there is to be showing

iProgramStuff replied:
Ah okay. I'll look at that component and see what it does when I drop it in here.

priscilawebdev commented on static/app/types/index.tsx:
```diff
@@ -1449,6 +1449,7 @@ export type ReleaseWithHealth = BaseRelease &
 type ReleaseData = {
   commitCount: number;
   data: {};
+  adoptionStages?: {};
```
should we also move this support to the end of this object since it's optional?

iProgramStuff replied:
Will do! :)

iProgramStuff committed:
Merge remote-tracking branch 'origin' into chrisf/releaseadoptionstate

iProgramStuff committed:
Putting stages behind flag, adding api test

iProgramStuff committed:
With adoption stages logic improvement

iProgramStuff committed:
Adding adoptionStages option to release details

iProgramStuff committed:
Merge remote-tracking branch 'origin' into chrisf/releaseadoptionstate

iProgramStuff committed:
Merge branch 'chrisf/releaseadoptionstate' into chrisf/feadoptionstage

robinrendle committed:
fix(ui): dark mode fixes for issue details page (#26933)

iProgramStuff committed:
Adding conditional column (attempt), using stage

Commit:
feat(integrations): Validate integration_id is int (#26934)

Objective:
Fixes SENTRY-PF0
Guard against invalid input for the integration_id path param.

iProgramStuff committed:
got column working

getsantry[bot] committed:
style(lint): Auto commit lint changes

iProgramStuff committed:
adding better type

getsantry[bot] committed:
style(lint): Auto commit lint changes

wedamija committed:
fix(rules): Add index to fix slow rule query (#26906)

https://getsentry.atlassian.net/browse/OPS-657

We're seeing this query take up a lot of cumulative time, this index should resolve it. Also materialized project_ids to make it easier for the planner to produce a good query.

Commit:
feat(ms-teams): Handle expired links (#26948)

Objective:
Fixes SENTRY-K8M
Currently, if a user tries to link their identity with an expired link, we just return the Internal Server Error page. This PR adds a template that we can return for this error case, so that the user can retry the linking process.

wedamija committed:
perf(inbox): Shave some complexity off of inbox queries. (#26950)

This pr materializes queries to fetch team ids related to a user, to help simplify the query plan.
This likely won't improve the query a lot, but having these complicated subqueries makes the
planner's job harder and can result in poor query plan choices sometimes.

taylangocmen committed:
fix(ui): Issue stream display options tooltips (#26947)

* fix(ui): Issue stream display options tooltips

* only show tooltip when % selected

* dont show tooltip when dropdown open

* button props type is required

evanpurkhiser committed:
ref(search): remove UNSAFE_componentWillReceiveProps (#26802)

evanpurkhiser committed:
feat(ui): Improve menuItem with icon + withBorder props (#26952)

Commit:
fix(appconnect): Detect UuidMap in dSYM zipfiles (#26925)

The PList/UUIDMap detection only worked if you supplied the DebugId as
is done in the assemble-difs endpoint.  However now that we ingest
dSYM files from Apple App Store Connect we do need to also handle
having to use the filename for the DebugId.

matejminar committed:
feat(ui): Add release comparison table (#26921)

matejminar committed:
fix(ui): Fix pagination of commits and files changed (#26959)

This will refetch commits/files when the cursor in the URL changes.

iProgramStuff committed:
feat(workflow): Adding adoption stage info to ReleaseSerializer (#26892)

iProgramStuff committed:
Adding actor context to feature check

iProgramStuff committed:
merging
